File: Bridges/PythonBridge.cs
using Python.Runtime;
using System;
using System.Collections.Generic;
using System.IO;

namespace OmnIDE.Bridges
{
    public class PythonBridge
    {
        public PythonBridge()
        {
            PythonEngine.Initialize();
        }

        private void AddPythonPath()
        {
            string pythonPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Python");

            using (Py.GIL())
            {
                dynamic sys = Py.Import("sys");
                sys.path.append(pythonPath);
            }
        }

        public bool CallDataProcessorExtractZip(string sourcePath, string destinationPath)
        {
            using (Py.GIL())
            {
                AddPythonPath();

                dynamic pyBridge = Py.Import("python_bridge");
                dynamic bridgeInstance = pyBridge.PythonBridge();

                dynamic result = bridgeInstance.call_data_processor_ExtractZip(sourcePath, destinationPath);

                return (bool)result;
            }
        }

        public Dictionary<string, object> CallCodeExecutor(List<string> files, Dictionary<string, object> metadata)
        {
            using (Py.GIL())
            {
                AddPythonPath();

                dynamic pyBridge = Py.Import("python_bridge");
                dynamic bridgeInstance = pyBridge.PythonBridge();

                dynamic result = bridgeInstance.call_code_executor(files.ToPython(), metadata.ToPython());

                return result.As<Dictionary<string, object>>();
            }
        }

        public Dictionary<string, object> CallReportGenerator(Dictionary<string, object> results)
        {
            using (Py.GIL())
            {
                AddPythonPath();

                dynamic pyBridge = Py.Import("python_bridge");
                dynamic bridgeInstance = pyBridge.PythonBridge();

                dynamic report = bridgeInstance.call_report_generator(results.ToPython());

                return report.As<Dictionary<string, object>>();
            }
        }

        public void Shutdown()
        {
            PythonEngine.Shutdown();
        }
    }
}

File: OmnIDEApi/Python/python_bridge.py
from DataProcessor import DataProcessor
from CodeExecutor import CodeExecutor
from ReportGenerator import ReportGenerator

class PythonBridge:
    def __init__(self):
        self.data_processor = DataProcessor()
        self.code_executor = CodeExecutor(code="")  # Provide a default code argument
        self.report_generator = ReportGenerator(report="")  # Provide a default report argument

    def call_data_processor(self, source_path, destination_path):
        return self.data_processor.extractZip(source_path, destination_path)

    def call_code_executor(self, files, metadata):
        return self.code_executor.run_code(files, metadata)

    def call_report_generator(self, results):
        return self.report_generator.generate_report(results)
